src/test/java/io/cronitor/client/CronitorPingerTest.java

```diff
@@ -0,0 +1,60 @@
+package io.cronitor.client;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.net.URL;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
+import java.net.HttpURLConnection;
+
+
+@PrepareForTest({ URL.class, HttpURLConnection.class, CronitorPinger.class })
+@RunWith(PowerMockRunner.class)
+public class CronitorPingerTest {
+
+
+    @Mock
+    private CronitorPinger pinger = new CronitorPinger();
+
+    private String code = "d3x0c1";
+    private String command = "run";
+    private String apiKey = null;
+    private String message = null;
+    private final static Logger logger = Logger.getLogger(CronitorPinger.class.getName());
+
+    @Test
+    public void canPingCronitorLink() throws Exception {
+        // set the url generated by passing usePrimaryPingDomain=true to setURL
+        URL url = pinger.getURL(true, command, code, apiKey, message);
+        pinger.setConnection(url);
+        pinger._ping();
+        assertEquals(pinger.connection.getResponseCode(), HttpURLConnection.HTTP_OK);
+        assertEquals(pinger.connection.getURL().toString(), "https://cronitor.link/d3x0c1/run");
+    }
+
+    @Test
+    public void canPingCronitorIO() throws Exception {
+        // set the url generated by passing usePrimaryPingDomain=false to setURL
+        URL url = pinger.getURL(false, command, code, apiKey, message);
+        pinger.setConnection(url);
+        pinger._ping();
+        assertEquals(pinger.connection.getResponseCode(), HttpURLConnection.HTTP_OK);
+        assertEquals(pinger.connection.getURL().toString(), "https://cronitor.io/d3x0c1/run");
+    }
+
+    @Test
+    public void shouldPingAgainAfterTimeout() throws Exception {
+        URL url = new URL("https://cronitor.link:81/d3x0c1/run");
+        PowerMockito.when(pinger.getURL(true, command, code, apiKey, message)).thenReturn(url);
```

Comment:
@lboix I'm having some trouble with this test set up in particular.

```org.mockito.exceptions.misusing.MissingMethodInvocationException: 
when() requires an argument which has to be 'a method call on a mock'.
For example:
    when(mock.getArticles()).thenReturn(articles);
Also, this error might show up because:
1. you stub either of: final/private/equals()/hashCode() methods.
   Those methods *cannot* be stubbed/verified.
   Mocking methods declared on non-public parent classes is not supported.
2. inside when() you don't call method on mock but on some other object.
	at io.cronitor.client.CronitorPingerTest.shouldPingAgainAfterTimeout(CronitorPingerTest.java:56)```

I'm guessing i'm just overlooking something in the set up. Any ideas?

Reply:
It seems that the call to the constructor needs to be removed, as the pinger has the annotation @Mock on it.